modules/custom/az_person/az_person.module

```diff
@@ -146,3 +148,40 @@ function az_person_theme() {
     'horizontal_tabs__node__az_person__group_tabs' => ['base hook' => 'horizontal_tabs'],
   ];
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * Prepopulate the alphabetical index letter field in az_person where available.
+ */
+function az_person_node_presave(EntityInterface $entity) {
+  if ($entity->bundle() === 'az_person'
+    && $entity instanceof NodeInterface
+    && $entity->hasField('field_az_alphabetical_index')
+    && $entity->hasField('field_az_lname')) {
```

Comment:
Opinionated suggestion to improve readability.
```suggestion
    && $entity->hasField('field_az_lname'))
  {
```
Apparently there Drupal's coding standards discourage breaking conditions into new lines currently but there's an open issue about relaxing that: https://www.drupal.org/project/coding_standards/issues/3392321